<launch>
  <!-- Export env variable so that gazebo finds the models and materials -->
  <env name="GAZEBO_MODEL_PATH" value="$(find labrob_worlds)/models" />
  
  <!-- We resume the logic in empty_world.launch, changing only the name of 
    the world to be launched -->
  <include file="$(find labrob_gazebo)/launch/labrob_gazebo.launch">
    <arg name="world_name" value="$(find labrob_worlds)/worlds/crazy_maze.world" />
    <arg name="spawn_args" value="-x 0 -y 0 -z 0"/>
    <arg name="show_rviz" value="true"/>
    <arg name="paused" value="false"/>
    <arg name="debug" value="false"/>  
    <arg name="gui" value="true"/>
  </include>
  
  <!-- Start the node that handles the map exploration percentage -->
  <node name="challenge_map_server" pkg="labrob_worlds" type="challenge_map_server.py" output="log">
    <param name="map_size_m2" value="400.0"/>
    <param name="map_resolution" value="0.05"/>
  </node>

</launch>
